File: src/main/kotlin/com/linqingying/cangjie/psi/CjSimpleNameStringTemplateEntry.java
package com.linqingying.cangjie.psi;

import com.linqingying.cangjie.psi.stubs.CangJiePlaceHolderWithTextStub;
import com.intellij.lang.ASTNode;
import org.jetbrains.annotations.NotNull;
import com.linqingying.cangjie.psi.stubs.elements.CjStubElementTypes;
import org.jetbrains.annotations.Nullable;

public class CjSimpleNameStringTemplateEntry extends CjStringTemplateEntryWithExpression {
    public CjSimpleNameStringTemplateEntry(@NotNull ASTNode node) {
        super(node);
    }

    public CjSimpleNameStringTemplateEntry(@NotNull CangJiePlaceHolderWithTextStub<CjSimpleNameStringTemplateEntry> stub) {
        super(stub, CjStubElementTypes.SHORT_STRING_TEMPLATE_ENTRY);
    }

    @Override
    public <R, D> R accept(@NotNull CjVisitor<R, D> visitor, @Nullable D data) {
        return visitor.visitSimpleNameStringTemplateEntry(this, data);
    }
}

File: src/main/kotlin/com/linqingying/cangjie/psi/CjSimpleNameStringTemplateEntry.kt
package com.linqingying.cangjie.psi

import com.linqingying.cangjie.psi.stubs.CangJiePlaceHolderWithTextStub
import com.linqingying.cangjie.psi.stubs.elements.CjStubElementTypes
import com.intellij.lang.ASTNode

class CjSimpleNameStringTemplateEntry : CjStringTemplateEntryWithExpression {
    constructor(node: ASTNode) : super(node)

    constructor(stub: CangJiePlaceHolderWithTextStub<CjSimpleNameStringTemplateEntry >) : super(
        stub,
        CjStubElementTypes.SHORT_STRING_TEMPLATE_ENTRY
    )

    override fun <R, D> accept(visitor: CjVisitor<R, D>, data: D?): R {
        return visitor.visitSimpleNameStringTemplateEntry(this, data)
    }
}
